contracts/axelar-gateway/src/test_gateway.rs

```diff
@@ -555,3 +555,56 @@ fn transfer_ownership_unauthorized() {
         }])
         .transfer_ownership(&new_owner);
 }
+
+#[test]
+fn version() {
+    let (env, _, client) = setup_env();
+    let owner = Address::generate(&env);
+    let operator = Address::generate(&env);
+
+    initialize(&env, &client, owner.clone(), operator, 1, randint(1, 10));
+
+    assert_eq!(
+        client.version(),
+        String::from_str(&env, env!("CARGO_PKG_VERSION"))
+    );
+}
+
+#[test]
+#[should_panic(expected = "HostError: Error(Storage, MissingValue)")]
+fn upgrade() {
+    let (env, _, client) = setup_env();
+    let owner = Address::generate(&env);
+    let operator = Address::generate(&env);
+    let new_wasm_hash = BytesN::<32>::from_array(&env, &[0; 32]);
+
+    initialize(&env, &client, owner.clone(), operator, 1, randint(1, 10));
+
+    // Should panic with invalid wasm hash
+    client.upgrade(&new_wasm_hash);
+}
+
+#[test]
+#[should_panic(expected = "HostError: Error(Auth, InvalidAction)")] // Unauthorized
+fn upgrade_unauthorized() {
+    let (env, contract_id, client) = setup_env();
+    let owner = Address::generate(&env);
+    let operator = Address::generate(&env);
+    let user = Address::generate(&env);
+    let new_wasm_hash = BytesN::<32>::from_array(&env, &[0; 32]);
+
+    initialize(&env, &client, owner.clone(), operator, 1, randint(1, 10));
+
+    assert_eq!(client.owner(), owner);
+    client
+        .mock_auths(&[MockAuth {
+            address: &user,
+            invoke: &MockAuthInvoke {
+                contract: &contract_id,
+                fn_name: "upgrade",
+                args: (new_wasm_hash.clone(),).into_val(&env),
+                sub_invokes: &[],
+            },
+        }])
+        .upgrade(&new_wasm_hash);
```

Comment:
Is there a positive test case for upgrade?

Reply:
I was trying to add a positive test, but it requires a new WASM hash which is obtained from `env.deployer().upload_contract_wasm(new_contract::WASM);` with `new_contract` imported from a WASM file. I'm trying to avoid using a file path here, and I also dug into the SDK, but no luck. I left a question on Discord and will share the answer once I get it.

Reply:
I got an answer from them: 
> Getting the contract Wasm Hash without specifying the file path isn't possible to do this at the moment. A native contract defined in the test/crate when registered doesn't have a wasm hash and can't be referenced by a wasm hash. In general the test enviroment doesn't support installing the contract, it's always registered as a contract instance.

Thus, I need to add an integration test with an actual Wasm file for a positive test. WIP.